pkg/kubelet/kuberuntime/kuberuntime_gc.go

```diff
@@ -338,9 +338,26 @@ func (cgc *containerGC) evictPodLogsDirectories(allSourcesReady bool) error {
 	logSymlinks, _ := osInterface.Glob(filepath.Join(legacyContainerLogsDir, fmt.Sprintf("*.%s", legacyLogSuffix)))
 	for _, logSymlink := range logSymlinks {
 		if _, err := osInterface.Stat(logSymlink); os.IsNotExist(err) {
-			err := osInterface.Remove(logSymlink)
+			firstlinkdstPath, err := os.Readlink(logSymlink)
 			if err != nil {
-				glog.Errorf("Failed to remove container log dead symlink %q: %v", logSymlink, err)
+				glog.Errorf("Failed to resolve the log symlink %q: %v", logSymlink, err)
+				return nil
+			}
+			secondlinkdstPath, err := os.Readlink(firstlinkdstPath)
+			if err != nil {
+				glog.Errorf("Failed to resolve the log firstlinkdstPath %q: %v", firstlinkdstPath, err)
+				return nil
+			}
+			dir := filepath.Dir(secondlinkdstPath)
+			if _, err := osInterface.Stat(dir); os.IsNotExist(err) {
+				err1 := osInterface.Remove(logSymlink)
+				if err1 != nil {
+					glog.Errorf("Failed to remove container log dead firstsymlink %q: %v", logSymlink, err1)
+				}
+				err2 := osInterface.Remove(firstlinkdstPath)
+				if err2 != nil {
+					glog.Errorf("Failed to remove container log dead secondsymlink %q: %v", firstlinkdstPath, err2)
```

Comment:
suggest : Failed to remove container log symlink, second symlink is not clear enough for normal user

Reply:
Yes，this seems much clear，I will take it，many thanks.